flytepropeller/pkg/compiler/transformers/k8s/workflow.go

```diff
@@ -159,6 +161,17 @@ func generateName(wfID *core.Identifier, execID *core.WorkflowExecutionIdentifie
 	}
 }
 
+const ExecutionIDSuffixLength = 21
```

Comment:
Can we make this a configurable value and if set to `0` (default?) then have it disabled (so no random characters are appended). My concern is that if anything relies on the CR name, this will break it.

Reply:
Yeah, I understand the config bloat all to well :). As you suggest, my main concern is breaking backwards compatibility here. I know there are Flyte users that rely on the FlyteWorkflow CR to be named identical to the execution ID, which this would break. For me to be comfortable merging this, I feel it should be defaulted to the current behavior. cc @eapolinario thoughts?

Reply:
Thanks for the feedback! I've updated the PR to address this and explain a bit more here: https://github.com/flyteorg/flyte/pull/5480#discussion_r1683582220